src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -566,43 +621,6 @@ bool MergeTask::ExecuteAndFinalizeHorizontalPart::prepare() const
     /// Merged stream will be created and available as merged_stream variable
     createMergedStream();
 
-    /// Skip fully expired columns manually, since in case of
-    /// need_remove_expired_values is not set, TTLTransform will not be used,
-    /// and columns that had been removed by TTL (via TTLColumnAlgorithm) will
-    /// be added again with default values.
-    ///
-    /// Also note, that it is better to do this here, since in other places it
-    /// will be too late (i.e. they will be written, and we will burn CPU/disk
-    /// resources for this).
-    if (!ctx->need_remove_expired_values)
-    {
-        auto part_serialization_infos = global_ctx->new_data_part->getSerializationInfos();
-
-        NameSet columns_to_remove;
```

Comment:
`columns_to_remove` is no longer used. Are they essentially the same as `global_ctx->new_data_part->expired_columns`? I'm curious why it was introduced in the first place.

Reply:
> Are they essentially the same as global_ctx->new_data_part->expired_columns

I think so.